notifications_utils/field.py

```diff
@@ -16,17 +18,38 @@ def __init__(self, body):
         # body shouldn’t include leading/trailing brackets, like (( and ))
         self.body = body.lstrip("(").rstrip(")")
 
+    class Types(StrEnum):
+        BASE = auto()
+        CONDITIONAL = auto()
+
+    extended_type_pattern = OrderedDict(
```

Comment:
I'd add a comment here about why the order of the items matters - that is it will determine the order we check for the types, and so the conditional should come first, from what I gather?